x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexing/AsyncTwoPhaseIndexer.java

```diff
@@ -178,6 +208,35 @@ public synchronized boolean maybeTriggerAsyncJob(long now) {
         }
     }
 
+    /**
+     * Cancels a scheduled search request and issues the search request immediately
+     */
+    private synchronized void stopThrottledSearch() {
+        if (scheduledNextSearch != null && scheduledNextSearch.cancel()) {
```

Comment:
From what I understand around `cancel`, the only times it will return `false` are:

- If the action has already been completed
- If the action has already been cancelled.

This means it will return `true` if the thread is executing.

```
threadPool.executor(executorName).execute(() -> checkState(getState()));
```
Could happen in the middle of `triggerNextSearch`. This MIGHT be ok, but the call to `checkState(getState())` might transition from `STOPPING -> STOPPED` while a search is still in flight. I am not sure this is intended behavior.